### ISSUE TITLE
mpm_worker support

### ISSUE THREAD

**Issue by jaybonci** [everything2/everything2#2424]
There are some interesting elements to support mpm_worker:

JSON::XS breaks, so we need to PerlSetEnv
PERL_JSON_BACKEND JSON::PP
Apache2::SizeLimit needs to be scoped to only prefork
There are some UTF-8 problems as well.

Will need to run performance tests to see if it actually saves memory or not.

**Comment by jaybonci** [comment]
mpm_worker / prefork is the process-model question the PSGI migration settles (Starman workers vs Apache prefork children). Folds into the epoch:psgi work; see psgi-plack-migration-plan.md.

**Comment by jaybonci** [comment]
## Reframe (2026-06-01): blocked on PSGI; this is a *consequence* of it, not standalone work

The original ask — run Apache under mpm_worker/event to save memory — is impossible while mod_perl is in the request path. `ModPerl::Registry` runs XS modules (JSON::XS, Cpanel::JSON::XS — loaded as in-process `PerlModule`s at apache2.conf.erb:277) *inside* each Apache child, and those aren't thread-safe under a threaded MPM. That's the "JSON::XS breaks" symptom in the original report. So prefork is currently a **conscious pin**, not an oversight: docker/e2app/Dockerfile:20 does `rm -f mpm_event.*` and :22 force-enables `mpm_prefork` + `perl` together.

**The flip happens as part of the PSGI migration (epoch:psgi), in one coordinated move:**
1. Remove mod_perl from Apache — drop `ModPerl::Registry`, `Apache::DBI`, `Apache2::compat`, and the in-process `PerlModule` loads (apache2.conf.erb:261-285). Perl execution moves to Starman.
2. Invert the Dockerfile MPM lines (docker/e2app/Dockerfile:20-22): stop removing `mpm_event`, enable `mpm_event` instead of `mpm_prefork`, drop the `perl` module.
3. Delete the `<IfModule mpm_prefork_module>` tuning block (apache2.conf.erb:204-209, StartServers/MaxRequestWorkers); worker count moves to Starman (`--workers ~10`).

Result: Apache front-end runs event MPM (cheap threads, parks keep-alives) as a pure proxy + security/static layer; Perl concurrency is Starman's bounded prefork pool. The memory win is Starman's worker count (120→~20), not the MPM itself. **Do not attempt before PSGI** — flipping the MPM with mod_perl loaded breaks the app. See docs/psgi-plack-migration-plan.md.
